### PR TITLE
Fix Homebrew deployment on CircleCI

### PR DESCRIPTION
**Pull request by ulope**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -549,18 +552,19 @@ jobs:
...
           command: |
             source .circleci/get_archive_tag.sh
-            UPDATED_SHA256=$(openssl dgst -sha256 dist/archive/raiden-${ARCHIVE_TAG}-macOS.zip)
-            FORMULA_FILE="homebrew-raiden/raiden.rb"
-            sed -i .bak -E "s/[0-9]+\.[0-9]+\.[0-9]+/${ARCHIVE_TAG/v/}/g" ${FORMULA_FILE}
-            sed -i .bak -E "s/sha256 \"[a-f0-9]\{64\}\"/sha256 \"${UPDATED_SHA256: -64}\"/g" ${FORMULA_FILE}
-            rm $FORMULA_FILE.bak
+            export UPDATED_SHA256=$(openssl dgst -sha256 dist/archive/raiden-${ARCHIVE_TAG}-macOS-x86_64.zip)
+            export FORMULA_FILE="homebrew-raiden/raiden.rb"
+            sed -i.bak -r "s/[0-9]+\.[0-9]+\.[0-9]+(-?rc.)?/${ARCHIVE_TAG/v/}/g" ${FORMULA_FILE}
```
Comment (rakanalh) [inline .circleci/config.yml]: Was removing the space in `-i.bak` done on purpose?

Comment (ulope) [reply .circleci/config.yml]: Yes it was. Previously (on Travis) this ran on macOS which has BSD sed. On Circle this is now running in a linux container with gnu sed which (for whatever reason) doesn't accept a space after the `-i` option.

**Comment by ulope** [comment]
> 1. Adding github.com to known_hosts because of `not in known_hosts` (IIRC) error

Yes, with additional checking of the key fingerprint to prevent MitM attacks.

> 2. Adding support to add RCs to homebrew

Also yes. Previously the `-rcX` suffixes would 'pile up' after multiple rc releases (e.g. `raiden-v0.100.2-rc2-rc3-macOS.zip`) since the regex didn't capture them.

Edit: Weird error in circle CI lint step... wondering how that happend 🤔

👀

**Comment by ulope** [comment]
Ok found the reason for the lint failure. On Circle we get `astroid==2.2.0` (as a dependency of pylint) which seems to cause this error. 

Related: PyCQA/astroid#651, PyCQA/pylint#2774

/edit: @konradkonrad already fixed it another way in #3557. Still we should pin astroid since it seems to break other stuff as well.

**Pull request merged**